version: '3.8'

services:
  server_app:
    build:
      context: ./server/docker/php-apache
      dockerfile: Dockerfile
    container_name: server_app
    restart: unless-stopped
    networks:
      - fullstack_challenge
    depends_on:
      - database
    ports:
      - '8080:80'
    volumes:
      - './server/app:/var/www/html'

  database:
    image: mysql:8.0
    container_name: database
    restart: unless-stopped
    environment:
      - MYSQL_DATABASE=${MYSQL_DATABASE}
      - MYSQL_ROOT_PASSWORD=${MYSQL_ROOT_PASSWORD}
      - MYSQL_USER=${MYSQL_USER}
      - MYSQL_PASSWORD=${MYSQL_PASSWORD}
    networks:
      - fullstack_challenge
    ports:
      - '3306:3306'
    volumes:
      - './server/docker/database/init.sql:/docker-entrypoint-initdb.d/init.sql'
      - 'fullstack_challenge_mysql_data:/var/lib/mysql'

  client_app:
    build:
      context: ./client
      dockerfile: ./docker/Dockerfile
    container_name: client_app
    restart: unless-stopped
    environment:
      NODE_ENV: development
    command: npm run dev
    networks:
      - fullstack_challenge
    depends_on:
      - server_app
    ports:
      - "3000:3000"
    volumes:
      - './client/app:/app'

networks:
  fullstack_challenge:

volumes:
  fullstack_challenge_mysql_data:
    external: false
